src/platform/forms-system/src/js/actions.js

```
@@ -114,7 +114,7 @@ export function submitToUrl(body, submitUrl, trackingPrefix, eventData) {
         // got this from the fetch polyfill, keeping it to be safe
         const responseBody =
           'response' in req ? req.response : req.responseText;
-        const results = JSON.parse(responseBody);
+        const results = JSON.parse(responseBody || '{}');
```

This is in the code path for every form's submission. Just spitballing, but could this change now be silencing error scenarios in a way that it shouldn't?

---

@nihil2501 just seeing your comment now... so I found that after we submitted the form the backend was returning a 204 with no response message and the frontend code was trying to parse the response message but it couldn't since there was no response message in the `responseBody` so it was causing a hard fail in the frontend and we couldnt tell what was happening. I was trying to fix hat issue but looking at this again I could see where you could be correct and this could be silencing errors...maybe we could check the status before resolving the results or something...what were your thoughts?